### ISSUE TITLE
Respect minimum maximum size of window

### ISSUE THREAD

**Issue by dim-geo** [lingtjien/Grid-Tiling-Kwin#19]
Hi, I have noticed that grid tiling does not respect minimum size imposed by the window (for example deluge) It would be nice if you could fix that!

**Comment by lingtjien** [comment]
That is correct, but I'm not sure if I want to implement this. Many applications have badly defined geometries and also save their geometries and have weird names and all that. My goal was to create a unified experience, by letting the user decide instead of the application. (Some applications also save the virtual desktop they were opened on last and then try to open on that specific virtual desktop for example...)

That's why I made the option to allow setting clients to a certain minSize so the user can define the minimum size a client can take up instead of the client itself.

Also, a traditional minimum and maximum size of a window are not very well translated into the minSize variable. As the actual minimum size of a client in this script is not defined by the minSize but also the divider bounds. 

In theory, I could at startup check the minimum and maximum size of windows and then set the minSize variable to something that would not go over these values but then I would also have to account for the dividers min and max to not cross these boundaries. 

So it would be a lot of extra checking in the script while the user can also just define the minSize of an application in the UI. This gives the user full control and also allows the user to define the minimum size a client takes up instead of the application.
